jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -142,8 +142,8 @@ public static boolean isSameName(Path pathA, Path pathB)
 
     PathResource(URI uri, boolean bypassAllowedSchemeCheck)
     {
-        // normalize to referenced location, Paths.get() doesn't like "/bar/../foo/text.txt" style references
-        // and will return a Path that will not be found with `Files.exists()` or `Files.isDirectory()`
+        // Normalize to referenced location, Paths.get() doesn't like "/bar/../foo/text.txt" style references
+        // and will return a Path that will not be found with `Files.exists()` or `Files.isDirectory()`.
         this(Paths.get(uri.normalize()), uri, bypassAllowedSchemeCheck);
```

Comment:
This `Paths.get(uri.normalize())` will both normalize and decode the URI. This means it could already be an alias for the true URI passed in.

Because of this we need the check at line 283 `else if (!URIUtil.equalsIgnoreEncodings(uri, toUri(path)))`.

I don't like this, but it makes you able to use the `PathResource.getPath()` to access the file, because otherwise using it you could get `FileNotFoundException` if it is an unnormalized path.

If we always used methods like `getTargetUri()` and `getTargetPath()` to access the file then this would not be an issue as we'd be accessing the resolved location of the resource.